pyhealth/datasets/omop.py

```diff
@@ -405,17 +408,38 @@ def measurement_unit(p_info):
 
             events = []
             for v_id, v_info in p_info.groupby("visit_occurrence_id"):
-                for timestamp, code in zip(
-                    v_info["measurement_datetime"], v_info["measurement_concept_id"]
+                for timestamp, code, value_numeric, value_categorical in zip(
+                    v_info["measurement_datetime"], v_info["measurement_concept_id"], v_info["value_as_number"], v_info["value_as_concept_id"]
                 ):
-                    event = Event(
-                        code=code,
-                        table=table,
-                        vocabulary="MEASUREMENT_CONCEPT_ID",
-                        visit_id=v_id,
-                        patient_id=p_id,
-                        timestamp=strptime(timestamp),
-                    )
+                    if not math.isnan(value_numeric):
+                        event = Event(
+                            code=code,
+                            table=table,
+                            vocabulary="MEASUREMENT_NUMERIC",
+                            visit_id=v_id,
+                            patient_id=p_id,
+                            timestamp=strptime(timestamp),
+                            measurement=value_numeric
+                        )
+                    elif value_categorical:
+                        event = Event(
+                            code=code,
+                            table=table,
+                            vocabulary="MEASUREMENT_CATEGORICAL",
+                            visit_id=v_id,
+                            patient_id=p_id,
+                            timestamp=strptime(timestamp),
+                            measurement=value_categorical
+                        )
+                    else: 
+                        event = Event(
+                            code=code,
+                            table=table,
+                            vocabulary="MEASUREMENT_CONCEPT_ID",
+                            visit_id=v_id,
+                            patient_id=p_id,
+                            timestamp=strptime(timestamp),
+                        )
```

Comment:
We will call the measurements codes using the vocabulary name, so they have to be the same, such as "MEASUREMENTS_CONCEPT_ID". Could we use the following?

```
event = Event(
    code=code,
    table=table,
    vocabulary="MEASUREMENTS_CONCEPT_ID",
    visit_id=v_id,
    patient_id=p_id,
    timestamp=strptime(timestamp),
    measurement=value_numeric if not math.isnan(value_numeric) elif value_categorical else: None
)
```